Intermittent unit test failure

```
CFLAGS="-DLIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT=1 -DLIBSPDM_FIPS_MODE=1 -DLIBSPDM_ENABLE_CAPABILITY_CSR_CAP_EX=1 -DLIBSPDM_DEBUG_LIBSPDM_ASSERT_CONFIG=3" cmake -G"NMake Makefiles" -DARCH=ia32 -DTOOLCHAIN=VS2019 -DTARGET=Debug -DCRYPTO=mbedtls
```
produced an access fault when testing the Requester. See https://github.com/DMTF/libspdm/actions/runs/17213773686/job/48832052945.
```
[ RUN      ] libspdm_test_requester_encap_request_case15
libspdm_send_spdm_request[0] msg SPDM_GET_ENCAPSULATED_REQUEST(0xea), size (0x4): 
0000: 11 ea 00 00 
libspdm_receive_spdm_response[0] msg SPDM_ENCAPSULATED_REQUEST(0x6a), size (0x28): 
[  ERROR   ] --- EXCEPTION_ACCESS_VIOLATION occurred at 6F397D57.

To debug in Visual Studio...
1. Select menu item File->Open Project
2. Change 'Files of type' to 'Executable Files'
3. Open this executable.
4. Select menu item Debug->Start

Alternatively, set the environment variable 
UNIT_TESTING_DEBUG to 1 and rebuild this executable, 
then click 'Debug' in the popup dialog box.


0000: 11 6a 00 00 11 81 00 01 f4 7a 8e c3 e9 af f2 31 8d 89 69 42 28 2a d4 fe 37 d6 39 1c 82 91 4f 54 
0020: a5 da 8a 37 de 13 00 c6 
[  FAILED  ] libspdm_test_requester_encap_request_case15
LIBSPDM_ASSERT: D:\a\libspdm\libspdm\unit_test\spdm_unit_test_common\common.c(138): !m_sender_buffer_acquired && !m_receiver_buffer_acquired
Error: Process completed with exit code 1.
```

--- a/unit_test/test_spdm_requester/encap_request.c
+++ b/unit_test/test_spdm_requester/encap_request.c
@@ -1307,6 +1307,8 @@ void libspdm_test_requester_encap_request_case14(void **State)
     spdm_context->local_context.capability.flags |= SPDM_GET_CAPABILITIES_REQUEST_FLAGS_CERT_CAP;
     spdm_context->connection_info.capability.flags |=
         SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ENCAP_CAP;
+    spdm_context->local_context.local_cert_chain_provision[0] = NULL;
+    spdm_context->local_context.local_cert_chain_provision_size[0] = 0;
     if (!libspdm_read_responder_public_certificate_chain(m_libspdm_use_hash_algo,
                                                          m_libspdm_use_asym_algo, &data,
                                                          &data_size,
@@ -1362,6 +1364,8 @@ void libspdm_test_requester_encap_request_case15(void **State)
     spdm_context->local_context.capability.flags |= SPDM_GET_CAPABILITIES_REQUEST_FLAGS_CERT_CAP;
     spdm_context->connection_info.capability.flags |=
         SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ENCAP_CAP;
+    spdm_context->local_context.local_cert_chain_provision[0] = NULL;
+    spdm_context->local_context.local_cert_chain_provision_size[0] = 0;
     if (!libspdm_read_responder_public_certificate_chain(m_libspdm_use_hash_algo,
                                                          m_libspdm_use_asym_algo, &data,
                                                          &data_size,

--- a/unit_test/test_spdm_requester/get_certificate.c
+++ b/unit_test/test_spdm_requester/get_certificate.c
@@ -4509,6 +4509,8 @@ void libspdm_test_requester_get_certificate_case30(void **state)
 
     free(data);
     free(m_libspdm_local_certificate_chain);
+    m_libspdm_local_certificate_chain = NULL;
+    m_libspdm_local_certificate_chain_size = 0;
 }
 
 /**
@@ -4592,6 +4594,11 @@ void libspdm_test_requester_get_certificate_case31(void **state)
                                      cert_chain);
     assert_int_equal(status, LIBSPDM_STATUS_BUFFER_TOO_SMALL);
     free(data);
+    if (m_libspdm_local_certificate_chain != NULL) {
+        free(m_libspdm_local_certificate_chain);
+        m_libspdm_local_certificate_chain = NULL;
+        m_libspdm_local_certificate_chain_size = 0;
+    }
 }
 
 int libspdm_requester_get_certificate_test_main(void)

